### PR TITLE
Add support for client port in audit record

### LINKED ISSUE #90: Include client port number in audit record
In C* 4.0 the client port number is included in the audit record. The port field is not present in all kinds of records which is indicated with a value of '0' (zero).

We should add support for this in ecAudit as well.

### PR DESCRIPTION
**Pull request by eperott**
Closes #90

### CODE REVIEW

**Comment by eperott** [comment]
Update fields in the updated compatibility matrix doc.

File: common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java [hunk 1]
```diff
@@ -135,104 +126,221 @@ public void testUnknownType() throws Exception
...
 
         assertThatExceptionOfType(IORuntimeException.class)
         .isThrownBy(() -> readMarshallable.readMarshallable(wireInMock))
-        .withMessageContaining("Corrupt client IP address field");
+        .withMessageContaining("Corrupt client_ip field");
     }
 
     @Test
-    public void testNullClientAddress() throws Exception
+    public void testNullClientAddress()
```
Comment (etedpet) [inline common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java]: Doesn't testIllegalClientAddress() and testNullClientAddress() test the same thing?
The exact code is being executed... The only difference is what happens in InetAddress.getByAddress()...
So we're more or less testing that method with these to different test cases.

Same goes for testIllegalCoordinatorAddress() / testNullCoordinatorAddress()

File: common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java [hunk 2]
```diff
@@ -41,55 +43,50 @@
...
+        assertThatRecordIsSame(actualAuditRecord, defaultValues);
     }
 
     @Test
     public void testReadBatchBatch() throws Exception
     {
-        givenNextRecordIs((short) 0, "ecaudit-batch", 42L, InetAddress.getByName("1.2.3.4").getAddress(), InetAddress.getByName("5.6.7.8").getAddress(), "john", UUID.fromString("bd92aeb1-3373-4d6a-b65a-0d60295f66c9"), Status.ATTEMPT.toString(), "Some operation");
-
-        AuditRecordReadMarshallable readMarshallable = new AuditRecordReadMarshallable();
+        RecordValues expectedValues = defaultValues
```
Comment (etedpet) [inline common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java]: Really liked this rewrite with the defaulValues.but...
Makes each test case short and readable, and it is clear how they differ from each other (not that much "noise"). 
Nice!

File: conf/audit.yaml [hunk 3]
```diff
@@ -37,7 +37,7 @@
 # Slf4jAuditLogger: Configure you logback.xml to specify location of log files and rotation policy. The following
 # optional parameters are accepted:
 # - log_format   - Format of the audit record sent to SLF4J. Fields can be configured with bash-style parameter
-#                  substitution. Supported parameters are ${CLIENT}, ${USER}, ${BATCH_ID}, ${STATUS}, ${OPERATION},
+#                  substitution. Supported parameters are ${CLIENT_IP}, ${USER}, ${BATCH_ID}, ${STATUS}, ${OPERATION},
```
Comment (etedpet) [inline conf/audit.yaml]: Add CLIENT_PORT and COORDINATOR_IP here...

File: doc/slf4j_logger.md [hunk 4]
```diff
@@ -69,7 +70,7 @@ was part of a batch or not. Also the TIMESTAMP field have a custom time format c
 logger_backend:
     - class_name: com.ericsson.bss.cassandra.ecaudit.logger.Slf4jAuditLogger
       parameters:
-      - log_format: "${TIMESTAMP}-> client=${CLIENT}, user=${USER}, status=${STATUS}, {?batch-id=${BATCH_ID}, ?}operation='${OPERATION}'"
+      - log_format: "${TIMESTAMP}-> client=${CLIENT_IP}, user=${USER}, status=${STATUS}, {?batch-id=${BATCH_ID}, ?}operation='${OPERATION}'"
         time_format: "yyyy-MM-dd HH:mm:ss.SSS"
         time_zone: UTC
```
Comment (etedpet) [inline doc/slf4j_logger.md]: The CHANGES.md file also needs update

File: common/src/main/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/AuditRecordReadMarshallable.java [hunk 5]
```diff
@@ -83,15 +84,29 @@ private String readType(WireIn wire) throws IORuntimeException
...
-    private InetAddress readAddress(WireIn wire, String key) throws IORuntimeException
+    private InetSocketAddress readInetSocketAddress(WireIn wire, String ipKey, String portKey) throws IORuntimeException
+    {
+        try
+        {
+            InetAddress inetAddress = InetAddress.getByAddress(wire.read(ipKey).bytes());
+            int port = wire.read(portKey).int32();
+            return new InetSocketAddress(inetAddress, port);
+        }
+        catch (UnknownHostException e)
```
[the remark this answers is not in the document]
Comment (eperott) [reply common/src/main/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/AuditRecordReadMarshallable.java]: Done

File: conf/audit.yaml (hunk 3 above)
Comment (eperott) [reply conf/audit.yaml]: Done

File: common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java [hunk 6]
```diff
@@ -41,55 +43,50 @@
 @RunWith(MockitoJUnitRunner.StrictStubs.class)
 public class TestAuditRecordReadMarshallable
 {
-    @Mock
+    @Mock(lenient = true)
```
[the remark this answers is not in the document]
Comment (eperott) [reply common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java]: Done

File: common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java (hunk 2 above)
Comment (eperott) [reply common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java]: Yeah, I thought so too. Updating a couple of other tests as well...

File: common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java (hunk 1 above)
Comment (eperott) [reply common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java]: True. But, but, but.... It feels right to check both these corner cases.... :-)
Wouldn't this add some value in a future refactoring job? On the other hand, if that's a valid argument, then I should have test cases for all other fields as well where I check for null and illegal data type and so forth...

File: doc/slf4j_logger.md (hunk 4 above)
Comment (eperott) [reply doc/slf4j_logger.md]: Done

File: common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java (hunk 1 above)
Comment (eperott) [reply common/src/test/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/TestAuditRecordReadMarshallable.java]: Never mind, removed the null versions.
